### ISSUE TITLE
NSInternalInconsistencyException - Directory server root certificates are invalid or nil

### ISSUE THREAD

**Issue by rokridi** [Adyen/adyen-3ds2-ios#34]
Hello.

I have noticed that the application am working on crashes when user issues a 3DS2 payment after upgrading the SDK to `2.3.3`.
The crash occurs only with PROD environment, no crash when testing with QA environment.

`Crashlytics` reports several crashes with this message:

`NSInternalInconsistencyException - Directory server root certificates are invalid or nil, providing the Directory server root certificates and public key is required. check [documentation](https://docs.adyen.com/online-payments/classic-integrations/api-integration-ecommerce/3d-secure/native
`

A sample code showing how i start a transaction:

```swift

public func transaction(
        serverIdentifier: String,
        publicKey: String,
        messageVersion: String,
        directoryServerRootCertificates: String,
        completion: @escaping (Result<any Adyen3DS2TransactionProtocol, any Error>) -> Void
    ) {
        let parameters = ADYServiceParameters(
            directoryServerIdentifier: serverIdentifier,
            directoryServerPublicKey: publicKey,
            directoryServerRootCertificates: directoryServerRootCertificates
        )
        self.adyService
            .service(parameters: parameters) { service in
                do {
                    try completion(.success(service.transaction(messageVersion: messageVersion)))
                } catch {
                    completion(.failure(error))
                }
            }
    }
```

 - Device: iPhone 7, iPhone 6s, iPhone 7 Plus, iPad Mini 4 ...
 - iOS:  15.8.0, 15.8.01, 17.3.0, 17.3.1

**Comment by robertdalmeida** [comment]
Hi @rokridi - thanks for reporting this issue. 
Could you please share a psp reference for this issue that you see?

**Comment by robertdalmeida** [comment]
Hi @rokridi, Thank you for sharing the psp. 
unfortunately, on investigation we find that for this psp we don't get the exception from the sdk and are unable to see the issue. (We are able to start the transaction based on the payload `ADYServiceParameters` that Adyen sends)

1. A possibility could be that your backend is filtering characters/any other sort of processing of these parameters when received from Adyen. Which in turn on being sent to the sdk fails. Could you please verify if this is case?
2. Could you send us a few more psps (to eliminate the possibility that there could be a scenario that the root certificates are nil/empty).

**Comment by rokridi** [comment]
Hello @robertdalmeida 
I will ask the backend team and answer you back asap.

**Comment by rokridi** [comment]
Here are some psp refereces:
- GGTJ4638LKV4J4C2
- GJ9Q52LZ6VKJBFG2
-  BXM6LQ6MR6KNQPB2
- WXLZP9MKWDKKFXB2
- FS95B4BC2Q29W953

I am still waiting for an answer from the backend team regarding you first qustion.

**Comment by rokridi** [comment]
The backend team says that the data received from Adyen backend is not processed before sending it to the front end.

**Comment by rokridi** [comment]
Hello @robertdalmeida 

Sorry I can't find any common pattern that systematically leads to reproducing this crash.
The crash doesn't occur in specific devices or iOS versions

**Comment by robertdalmeida** [comment]
Hi @rokridi, Thanks for checking. We are working on an updated release of the sdk. Should be available in a week's time.

**Comment by ggdiez** [comment]
I'm having exactly the same crashes reports on `Crashlytics` 

```
Fatal Exception: NSInternalInconsistencyException
Directory server root certificates are invalid or nil, providing the Directory server root certificates and public key is required. check [documentation](https://docs.adyen.com/online-payments/classic-integrations/api-integration-ecommerce/3d-secure/native
```

Waiting for the next version.

**Comment by ggdiez** [comment]
Hi @robertdalmeida I see that you published new version 2.4.1 and I guess the fix is inside. 
Do you know when adyen-ios will publish a new version pointed to 2.4.1 of this repo?

**Comment by robertdalmeida** [comment]
@ggdiez the version from `adyen-ios` is planned for release around march end.

**Comment by robertdalmeida** [comment]
Closing this issue as the fix for this has been released in version 2.4.1
